Sources/ArgumentParser/Parsing/CommandParser.swift

```diff
@@ -142,35 +142,48 @@ extension CommandParser {
     // Build the argument set (i.e. information on how to parse):
     let commandArguments = ArgumentSet(currentNode.element, visibility: .private)
     
-    // Parse the arguments, ignoring anything unexpected
-    let values = try commandArguments.lenientParse(
-      split,
-      subcommands: currentNode.element.configuration.subcommands,
-      defaultCapturesAll: currentNode.element.defaultIncludesUnconditionalArguments)
-
-    // Decode the values from ParsedValues into the ParsableCommand:
-    let decoder = ArgumentDecoder(values: values, previouslyDecoded: decodedArguments)
-    var decodedResult: ParsableCommand
-    do {
-      decodedResult = try currentNode.element.init(from: decoder)
-    } catch let error {
-      // If decoding this command failed, see if they were asking for
-      // help before propagating that parsing failure.
-      try checkForBuiltInFlags(split)
-      throw error
+    var values: ParsedValues?
```

Comment:
Let's see if we can avoid making `values`, `decodedResult`, and `decoder` optional here — is that easier to manage if you refactor these loops out into functions?

Reply:
done